src/dbnode/storage/index/block.go

```diff
@@ -253,30 +258,74 @@ func (b *block) Query(
 		return false, err
 	}
 
-	var (
-		size       = results.Size()
-		brokeEarly = false
-	)
+	size := results.Size()
+	limitedResults := false
```

Comment:
can you just call this exhaustive and return it at the end instead of doing the limitedResults -> exhaustive translation

Reply:
Reads worse, it was actually like this before.